ocs/ocp.py

```diff
@@ -71,7 +72,12 @@ def exec_oc_cmd(self, command):
         except ValueError:
             pass
 
-        return yaml.safe_load(out)
+        try:
+            return yaml.safe_load(out)
+        except ScannerError:
+            # Exception could be because of output from normal commands like
+            # Fio for which yaml.safe_load() fails
```

Comment:
What if that's not the reason for the exception? Can you give an example?

This will affect all callers of this function, so perhaps it's not a great idea to make this behavior the default. I would also maybe encourage some logging here.

Reply:
At least in case of Fio that's what I was hitting, output of fio looks something like 
```                                                     fio-rand-write: (groupid=0, jobs=1): err= 0: pid=561: Wed Jun 26 21:34:30 2019
                                                       write: io=1040.3MB, bw=106500KB/s, iops=26625, runt= 10002msec
                                                         slat (usec): min=2, max=2157.7K, avg=27.39, stdev=4242.51
                                                         clat (usec): min=1, max=37036, avg= 3.90, stdev=154.11
                                                          lat (usec): min=4, max=2157.7K, avg=31.29, stdev=4245.32
                                                         clat percentiles (usec):
                                                          |  1.00th=[    1],  5.00th=[    1], 10.00th=[    1], 20.00th=[    2],
                                                          | 30.00th=[    2], 40.00th=[    2], 50.00th=[    2], 60.00th=[    2],
                                                          | 70.00th=[    2], 80.00th=[    2], 90.00th=[    2], 95.00th=[    3],
                                                          | 99.00th=[    3], 99.50th=[    3], 99.90th=[   20], 99.95th=[   63],
                                                          | 99.99th=[ 8032]
                                                         lat (usec) : 2=15.08%, 4=84.53%, 10=0.10%, 20=0.19%, 50=0.05%
                                                         lat (usec) : 100=0.01%, 250=0.01%, 500=0.01%, 750=0.01%, 1000=0.01%
                                                         lat (msec) : 2=0.01%, 4=0.01%, 10=0.01%, 20=0.01%, 50=0.01%
                                                       cpu          : usr=12.89%, sys=15.97%, ctx=1768, majf=0, minf=10 
```
for which yaml.safe_load() fails with ScannerError . I thought if its safe_loadable output then we are good at yaml.safe_load() if not then just pass the output. I am not sure any other kind of errors which may pop up.
Do you feel in case of exception just return ```out``` ?

Reply:
It's hard to answer without a full (entire yaml) object. Would you put one in http://pastebin.test.redhat.com/ and link it please?

Reply:
@zmc 
The output from fio command is here https://paste.fedoraproject.org/paste/5GP4j50mrVlbgSGBm38nPQ. When yaml.safe_load() tried on the output string it throws ScannaerError.

Reply:
I dont think we need to convert the ouptut here, we should just pass it as is?

Reply:
Yeah, that doesn't look like yaml at all

Reply:
Why did you dismiss this, @vasukulkarni ?

Reply:
@zmc thats a different issue, I am going to fix that https://github.com/red-hat-storage/ocs-ci/issues/312

Reply:
@vasukulkarni ok, I'll put my comments in that issue instead of here.